### PR TITLE
Limiting file size

### PR DESCRIPTION
**Pull request by SoujitD-SAP**
Limit the upload size of an attachment in the attachment plugin. Currently the maximum size size of data that can be scanned by the Malware Scanning Service on BTP is 400MB.  So we would limit the file upload size to 400 MB.

1. Handling PUT request with 'before' handler to fetch the content length of the uploaded attachment in plugin.js.

2.  Impose validation checks for size limit and rejecting the request which are above size limit.

3. Currently cannot access the UI part but validation checks are both imposed for hybrid and local mode.

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -56,8 +57,36 @@ cds.once("served", async function registerPluginHandlers() {
     }
   }
 
+  function computeAttachment(req) {
```
Comment (muskansethi1) [inline lib/plugin.js:60]: rename the function, something like checkAttachmentLength?

File: lib/plugin.js [hunk 2]
```diff
@@ -41,6 +41,7 @@ cds.once("served", async function registerPluginHandlers() {
             srv.before("READ", [target, target.drafts], validateAttachment);
 
             srv.after("READ", [target, target.drafts], readAttachment);
+            srv.before("PUT", [target, target.drafts], computeAttachment);
```
Comment (muskansethi1) [inline lib/plugin.js:44]: do we need to do this in target too?

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+## Version 1.1.9
```
Comment (muskansethi1) [inline CHANGELOG.md:7]: Changelog has to be added in 1.1.8 version itself

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
...
+
+### Added
+
+- Size validation check for uploaded attachments.
```
Comment (muskansethi1) [inline CHANGELOG.md:11]: `Added limit for file size ....`

File: lib/plugin.js [hunk 5]
```diff
@@ -57,34 +61,60 @@ cds.once("served", async function registerPluginHandlers() {
...
     });
 
-    if(req?.req?.url?.endsWith("/content")) {
+    if (req?.req?.url?.endsWith("/content")) {
       const attachmentID = req.req.url.match(attachmentIDRegex)[1];
-      const status = await AttachmentsSrv.getStatus(req.target, { ID : attachmentID });
-      const scanEnabled = cds.env.requires?.attachments?.scan ?? true
-      if(scanEnabled && status !== 'Clean') {
-        req.reject(403, 'Unable to download the attachment as scan status is not clean.');
+      const status = await AttachmentsSrv.getStatus(req.target, {
```
Comment (muskansethi1) [inline lib/plugin.js:75]: you can remove the code formatting

File: lib/plugin.js [hunk 6]
```diff
@@ -57,34 +61,60 @@ cds.once("served", async function registerPluginHandlers() {
...
+          "Unable to download the attachment as scan status is not clean."
+        );
       }
     }
   }
 
   async function readAttachment([attachment], req) {
-    if (!req?.req?.url?.endsWith("/content") || !attachment || attachment?.content) return;
-    let keys = { ID : req.req.url.match(attachmentIDRegex)[1]};
+    if (
```
Comment (muskansethi1) [inline lib/plugin.js:89]: here also

**Comment by muskansethi1** [comment]
You can edit the script in package.json - `"test": "npx jest attachments.test.js"` to include unit testing

**Review by muskansethi1 — APPROVED** [review]

**Comment by SoujitD-SAP** [comment]
Changes in CHANGELOG.md with mention of UI5 version compatibilty. Tested in local and hybrid mode.

File: lib/plugin.js [hunk 7]
```diff
@@ -6,7 +6,7 @@ const attachmentIDRegex = /\/\w+\(.*ID=([0-9a-fA-F-]{36})/;
 
 cds.on("loaded", function unfoldModel(csn) {
   if (!("Attachments" in csn.definitions)) return;
-  const csnCopy = structuredClone(csn)
+  const csnCopy = structuredClone(csn);
```
Comment (jeevitha011) [inline lib/plugin.js:9]: Can you remove all the formatting changes in this file, as these lines are out of scope for this specific task.

**Review by jeevitha011 — APPROVED** [review]

**Comment by SoujitD-SAP** [comment]
Formatted by tallying from main branch

**Review by jeevitha011 — APPROVED** [review]
